package/src/nvidiamon.cpp

```diff
@@ -38,18 +38,51 @@ nvidiamon::nvidiamon()
   valid = test_nvidia_smi();
 }
 
-void nvidiamon::update_stats(const std::vector<pid_t>& pids) {
+std::pair<int, std::vector<std::string>> nvidiamon::read_gpu_stats_test(
+    const std::string read_path) {
+  std::vector<std::string> split_output{};
+  std::string output;
+  std::pair<int, std::vector<std::string>> ret{0, split_output};
+  const size_t buf_len = 100;
+  char buffer[buf_len];
+  FILE* inp = fopen(read_path.c_str(), "r");
+  while (fgets(buffer, buf_len, inp) != NULL) {
```

Comment:
Here it would be much better to use `std::getline`, which should read entire lines at a time, up to the newline.

The reason that the more complex method was used in `prmon::cmd_pipe_output` is that when reading from a *pipe* the process that is being read from can produce output sporadically, causing `getline` to return partial lines. Here the read is from a file, so it's going to work just fine with `getline`.